budget/app.py

```diff
@@ -27,18 +29,13 @@ def get_users(teams):
 
   Returns a dictionary of users with a list of their team memberships
   '''
+  syn = synapseclient.Synapse()
   teams_by_user_id = {}
   for team_id in teams:
-    synapse_url = configuration.get_synapse_team_member_url(team_id)
-    response = requests.get(synapse_url)
-    if response.ok:
-      results = json.loads(response.text)['results']
-      user_ids = [
-        result['member']['ownerId']
-        for result in results if not result['isAdmin']
-      ]
-    else:
-      response.raise_for_status()
+    user_ids = [
+      result['member']['ownerId']
+      for result in syn.getTeamMembers(team_id) if not result['isAdmin']
```

Comment:
I would recommend adding soe error handling, handling unexpected conditions, and clarifying messaging here. Under what circumstances will the synapseclient show an error? Will the synapseclient still return an empty list if the team id is not found?

Reply:
>  would recommend adding soe error handling, handling unexpected conditions, and clarifying messaging here.

Can you elaborate on this request?  In particular, why is additional error handling needed for `Synapse.getTeamMembers()` and not for `requests.get()` which it replaces?

 > Will the synapseclient still return an empty list if the team id is not found?

After https://sagebionetworks.jira.com/browse/PLFM-6390 is done it will throw an error.    The behavior of Synapse.getTeamMembers()` and `requests.get()` is the same in this regard.

Reply:
The old code using requests actually had error handling, `if response.ok ...else`. The other suggestions I was making were improvements. It is good to know that getTeamMembers will throw an error, too.

Reply:
I see.   I can confirm that all the Synapse programmatic client, (Python, R, and Java) convert non-2xx statuses to exceptions, so there is no status checked required in any app that uses one of those clients.

Thanks for suggesting the improvements.